components/calendar/benches/date.rs

```diff
@@ -0,0 +1,165 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+mod fixtures;
+
+use crate::fixtures::structs::DateFixture;
+use criterion::{
+    black_box, criterion_group, criterion_main, measurement::WallTime, BenchmarkGroup, Criterion,
+};
+use icu_calendar::{
+    iso::IsoDay, iso::IsoMonth, iso::IsoYear, AsCalendar, Calendar, Date, DateDuration,
+};
+
+fn bench_dates<A: AsCalendar>(dates: Vec<&mut Date<A>>) {
+    for date in dates {
+        // black_box used to avoid compiler optimization.
+        // Arithmetic
+        date.add(DateDuration::new(
+            black_box(1),
+            black_box(2),
+            black_box(3),
+            black_box(4),
+        ));
+
+        // Retrieving vals
+        let _ = black_box(date.year().number);
+        let _ = black_box(date.month().number);
+        let _ = black_box(date.day_of_month().0);
+
+        // Conversion to ISO.
+        let _ = black_box(date.to_iso());
+    }
+}
+
+#[allow(dead_code)]
```

Comment:
Suggestion: `#[cfg(feature = "bench")]` instead of `#[allow(dead_code)]`

Reply:
Done